dali/operators/random/normal_distribution_cpu.cc

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/random/rng_base_cpu.h"
+#include "dali/operators/random/normal_distribution.h"
+#include "dali/pipeline/operator/arg_helper.h"
+
+namespace dali {
+
+DALI_SCHEMA(random__NormalDistribution)
+    .DocStr(R"code(Generates random numbers following a normal distribution.
+
+The shape of the generated data can be either specified explicitly with a ``shape`` argument,
+or chosen to match the shape of the input, if provided. If none are present, a single number is
+generated.
+)code")
+    .NumInput(0, 1)
+    .NumOutput(1)
+    .AddOptionalArg<float>("mean",
+      R"code(Mean of the distribution.)code",
+      0.f, true)
+    .AddOptionalArg<float>("stddev",
+      R"code(Standard deviation of the distribution.)code",
+      1.f, true)
+    .AddParent("RNGAttr");
+
+class NormalDistributionCPU : public NormalDistribution<CPUBackend, NormalDistributionCPU> {
+ public:
+  template <typename T>
+  struct Dist {
+    using FloatType =
+      typename std::conditional<
+          ((std::is_integral<T>::value && sizeof(T) > 3) || sizeof(T) > 4),
```

Comment:
Why this 3? Can be just >=4? Even if this is the same it would be more readable.

Reply:
It can be >= 4

Reply:
`>= 4` seems to be more readable, as 3 is rather infrequent value for sizeof.